telegram/_bot.py

```diff
@@ -188,6 +195,29 @@ def __init__(
                 private_key, password=private_key_password, backend=default_backend()
             )
 
+    def __eq__(self, other: object) -> bool:
+        return self.bot == other
+
+    def __hash__(self) -> int:
+        return hash(self.bot)
+
+    def __reduce__(self) -> NoReturn:
+        """Called by pickle.dumps(). Serializing bots is unadvisable, so we forbid pickling."""
+        raise pickle.PicklingError('Bot objects cannot be pickled!')
+
+    def __deepcopy__(self, memodict: dict) -> 'Bot':
+        """Bot objects can't be deepcopied since they are not pickable, so just return self."""
+        return self
+
+    def to_dict(self) -> JSONDict:
+        """See :meth:`telegram.TelegramObject.to_dict`."""
+        data: JSONDict = {'id': self.id, 'username': self.username, 'first_name': self.first_name}
+
+        if self.last_name:
+            data['last_name'] = self.last_name
```

Comment:
This line was not covered by tests. Can you please either add a test case or revert the moving of this method?